crates/router/src/services/api/client.rs

```diff
@@ -321,6 +342,35 @@ impl ApiClient for ProxyClient {
             inner: Some(client_builder.request(method, url)),
         }))
     }
+    async fn send_request(
+        &self,
+        state: &AppState,
+        request: Request,
+        option_timeout_secs: Option<u64>,
+        _forward_to_kafka: bool,
+    ) -> CustomResult<reqwest::Response, ApiClientError> {
+        crate::services::send_request(state, request, option_timeout_secs).await
+    }
+
+    fn add_request_id(&mut self, _request_id: Option<String>) {
+        self.request_id = _request_id
+    }
+
+    fn get_request_id(&self) -> Option<String> {
+        self.request_id.clone()
+    }
+
+    fn add_merchant_id(&mut self, _merchant_id: Option<String>) {}
+
+    fn get_merchant_id(&self) -> Option<String> {
+        None
+    }
```

Comment:
where is this needed ??

Reply:
Removed this function and also `get_flow_name()` .